domain/application/service/application.go

```diff
@@ -782,10 +786,81 @@ func (s *ApplicationService) GetApplicationScalingState(ctx context.Context, app
 	}, errors.Trace(err)
 }
 
+// AgentVersionGetter is responsible for retrieving the agent version for a
+// given model.
+type AgentVersionGetter interface {
+	// GetModelAgentVersion returns the agent version for the specified model.
+	// If the agent version cannot be found, an error satisfying
+	// [modelerrors.NotFound] will be returned.
+	GetModelAgentVersion(context.Context, coremodel.UUID) (version.Number, error)
+}
+
+// Provider defines the interface for interacting with the underlying model
+// provider.
+type Provider interface {
+	environs.SupportedFeatureEnumerator
+}
+
+// ProviderApplicationService defines a service for interacting with the underlying
+// model state.
+type ProviderApplicationService struct {
+	ApplicationService
+
+	modelID            coremodel.UUID
```

Comment:
Are we passing this in via the service factory or pulling it off the DB every time? Confused which method we should be taking now?

Reply:
Model id is service factory.